src/domain/flow-system/domain/src/entities/flow/flow_state.rs

```diff
@@ -143,6 +135,19 @@ impl Projection for FlowState {
                             Ok(FlowState { triggers, ..s })
                         }
                     }
+                    E::Enqueued(FlowEventEnqueued { enqueued_for, .. }) => {
+                        if s.outcome.is_some() || s.timing.awaiting_executor_since.is_some() {
+                            Err(ProjectionError::new(Some(s), event))
+                        } else {
+                            Ok(FlowState {
+                                timing: FlowTimingRecords {
+                                    enqueued_for: Some(enqueued_for),
```

Comment:
Does this need to be reset somewhere? E.g. if flow is cancelled do we want to clear this field in the projected state?

Reply:
No, we don't need to. This is always used in combination with status, and later remains for history.
We don't clear timing records too, like when the task was scheduled.